### PR TITLE
installer: Correct SOS flashing for Pi Zero and Pi Zero W

### PR DESCRIPTION
**Pull request by dinosore**
On the Pi Zero and Zero W, 0 turns the led on and 1 turns it off.  led0 is the activity led on the Zeros.

This is the opposite way round to other models.

I have tested the code on a Pi Zero W and a Model B;

### CODE REVIEW

File: scripts/opt/raspberrypi-ua-netinst/install.sh [hunk 1]
```diff
@@ -8,6 +8,8 @@ variables_reset() {
 	final_action=
 	rpi_hardware=
 	rpi_hardware_version=
+	led_on=
+	led_off=
```
Comment (FooDeas) [inline scripts/opt/raspberrypi-ua-netinst/install.sh]: Please move them into the `led_sos()` function and declare them `local`.

File: scripts/opt/raspberrypi-ua-netinst/install.sh [hunk 2]
```diff
@@ -169,29 +171,29 @@ led_sos() {
 	if [ -e /sys/class/leds/led0 ]; then (echo none > /sys/class/leds/led0/trigger) &> /dev/null; else led0=; fi
 	if [ -e /sys/class/leds/led1 ]; then (echo none > /sys/class/leds/led1/trigger) &> /dev/null; else led1=; fi
 	for i in $(seq 1 3); do
-		if [ -n "$led0" ]; then (echo 1 > "${led0}"/brightness) &> /dev/null; fi
-		if [ -n "$led1" ]; then (echo 1 > "${led1}"/brightness) &> /dev/null; fi
+		if [ -n "$led0" ]; then (echo $led_on > "${led0}"/brightness) &> /dev/null; fi
+		if [ -n "$led1" ]; then (echo $led_on > "${led1}"/brightness) &> /dev/null; fi
```
Comment (FooDeas) [inline scripts/opt/raspberrypi-ua-netinst/install.sh:182]: `${led_on}` and `${led_off}`

File: scripts/opt/raspberrypi-ua-netinst/install.sh [hunk 3]
```diff
@@ -603,6 +605,15 @@ case "${rpi_hardware}" in
...
 
+# Setting leds on and off works the other way round on Pi Zero and Pi Zero W
+if [ ${rpi_hardware_version:0:4} = "Zero" ]; then
+	led_on=0
+	led_off=1
+else
+	led_on=1
+	led_off=0
+fi
+
```
Comment (FooDeas) [inline scripts/opt/raspberrypi-ua-netinst/install.sh]: Move into the sos function, too.

**Review by FooDeas — CHANGES_REQUESTED** [review]

**Comment by dinosore** [comment]
done

**Review by FooDeas — APPROVED** [review]

**Comment by FooDeas** [comment]
Thanks a lot!

**Pull request merged**